Src/Base/AMReX_iMultiFab.cpp

```diff
@@ -338,12 +342,14 @@ iMultiFab::sum (int comp, int nghost, bool local) const
             reduce_op.eval(bx, reduce_data,
             [=] AMREX_GPU_DEVICE (int i, int j, int k) -> ReduceTuple
             {
-                return { static_cast<Long>(arr(i,j,k,comp)) };
+                return { static_cast<unsigned long long>(arr(i,j,k,comp))+imax };
             });
+            points += bx.numPts();
         }
 
         ReduceTuple hv = reduce_data.value();
-        sm = amrex::get<0>(hv);
+        ulsm = amrex::get<0>(hv) - points*imax;
+        sm = static_cast<Long>(ulsm);
```

Comment:
I don't think this is correct. Because `ulsm` is unsigned, it can never be negative after the cast.

Reply:
You have to cast `amrex::get<0>(hv)` and `points*imax` to `long long` first.

Reply:
Good catch. I'm currently testing:
```
        long long ulsm = static_cast<long long>(amrex::get<0>(hv)) 
                       - static_cast<long long>(INT_MAX)*points;
```
I think this'll work, given we expect the final answer to be a `long`.

Reply:
Assuming you are using `{ static_cast<unsigned long long>(arr(i,j,k,comp)) - static_cast<unsigned long long>(INT_MIN) }` in the kernel, I think this should be `static_cast<long long>(amrex::get<0>(hv)) + static_cast<long long>(INT_MIN)*points` and you want `points` to be `long long`, not unsigned.